File: backend/app/schemas/map_config.py
from pydantic import BaseModel
from enum import Enum

class MapSource(str, Enum):
    OVERTURE = "overture"
    OSM = "osm"

class LayerType(str, Enum):
    BUILDINGS = "buildings"
    ROADS = "roads"
    LANDUSE = "landuse"

class MapStyleConfig(BaseModel):
    background_color: str = "#f0f0f0"
    building_color: str = "#cccccc"
    road_color: str = "#ffffff"
    water_color: str = "#a0c8f0"

class LayerConfig(BaseModel):
    show_buildings: bool = True
    show_roads: bool = True
    show_landuse: bool = False

class MapConfigRequest(BaseModel):
    map_source: MapSource
    layers: LayerConfig
    style: MapStyleConfig
    bbox: tuple[float, float, float, float]  # [min_lon, min_lat, max_lon, max_lat]

class TileConfigResponse(BaseModel):
    tile_url: str
    style_json: dict
    layers_enabled: list[LayerType]

File: frontend/src/types/mapConfig.ts
export const DataSource = {
  OVERTURE: 'overture',
  OSM: 'osm'
} as const;

export type DataSource = typeof DataSource[keyof typeof DataSource];

export const LayerType = {
  BUILDINGS: 'buildings',
  ROADS: 'roads',
  LANDUSE: "landuse"
} as const;

export type LayerType = typeof LayerType[keyof typeof LayerType];

export interface MapStyleConfig {
  background_color: string;
  building_color: string;
  road_color: string;
  water_color: string;
}

export interface LayerConfig {
  show_buildings: boolean;
  show_roads: boolean;
  show_landuse: boolean;
}

export interface MapConfigRequest {
  map_source: DataSource;
  layers: LayerConfig;
  style: MapStyleConfig;
  bbox: [number, number, number, number]; // [min_lon, min_lat, max_lon, max_lat]
}

export interface TileConfigResponse {
  tile_url: string;
  style_json: Record<string, any>;
  layers_enabled: LayerType[];
}